libmozdata/fx_trains.py

```diff
@@ -14,10 +14,26 @@ class FirefoxTrains:
     URL = "https://whattrainisitnow.com/api/"
     TIMEOUT = 30
 
+    _instance = None
+
+    def __init__(self) -> None:
+        """Constructor"""
+        self._cache = {}
+
+    @classmethod
+    def get_instance(cls):
+        """Get the singleton instance of FirefoxTrains."""
+        if cls._instance is None:
+            cls._instance = cls()
+        return cls._instance
```

Comment:
Nit: you could have `_cache` be a static attribute so you don't have to implement singleton

Reply:
Yes, but with that the cache will be shared across all instances. This is not a problem for our use cases in BugBot. But in general, this behaviour could be confusing.

I think even the caching should be explicit (i.e., to be enabled using a constructor parameter) to avoid unexpected magical behaviour.

Reply:
This WFM!